src/Orleans.Reminders/ReminderService/LocalReminderService.cs

```diff
@@ -25,41 +34,108 @@ internal sealed class LocalReminderService : GrainService, IReminderService
 
-        internal LocalReminderService(
-            Silo silo,
+
+        private readonly string typeName;
+
+        /// <summary>Gets the token for signaling cancellation upon stopping of grain service</summary>
+        private CancellationTokenSource StoppedCancellationTokenSource { get; }
+
+        private ReminderServiceStatus Status
+        {
+            get { return status; }
+            set
+            {
+                OnStatusChange(status, value);
+                status = value;
+            }
+        }
+
+        /// <summary>Gets the monotonically increasing serial number of the version of the ring range owned by the grain service instance</summary>
+        private int RangeSerialNumber { get; set; }
+
+        /// <summary>Gets the range of the partitioning ring currently owned by the grain service instance</summary>
+        private IRingRange RingRange { get; set; }
+
+        public LocalReminderService(
+            ILocalSiloDetails silo,
             IReminderTable reminderTable,
-            TimeSpan initTimeout,
+            IOptions<ReminderOptions> reminderOptions,
             ILoggerFactory loggerFactory,
-            IAsyncTimerFactory asyncTimerFactory)
-            : base(SystemTargetGrainId.CreateGrainServiceGrainId(GrainInterfaceUtils.GetGrainClassTypeCode(typeof(IReminderService)), null, silo.SiloAddress), silo, loggerFactory)
+            IAsyncTimerFactory asyncTimerFactory,
+            IConsistentRingProvider consistentRingProvider)
         {
+            this.StoppedCancellationTokenSource = new CancellationTokenSource();
+
+            this.siloAddress = silo.SiloAddress;
+            this.typeName = this.GetType().FullName;
             this.reminderTable = reminderTable;
-            this.initTimeout = initTimeout;
+            this.reminderOptions = reminderOptions;
+            this.initTimeout = reminderOptions.Value.InitializationTimeout;
             this.asyncTimerFactory = asyncTimerFactory;
             tardinessStat = AverageTimeSpanStatistic.FindOrCreate(StatisticNames.REMINDERS_AVERAGE_TARDINESS_SECONDS);
             IntValueStatistic.FindOrCreate(StatisticNames.REMINDERS_NUMBER_ACTIVE_REMINDERS, () => localReminders.Count);
             ticksDeliveredStat = CounterStatistic.FindOrCreate(StatisticNames.REMINDERS_COUNTERS_TICKS_DELIVERED);
             startedTask = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
             this.logger = loggerFactory.CreateLogger<LocalReminderService>();
-            this.listRefreshTimer = asyncTimerFactory.Create(Constants.RefreshReminderList, "ReminderService.ReminderListRefresher");
+            this.listRefreshTimer = asyncTimerFactory.Create(this.reminderOptions.Value.RefreshReminderListPeriod, "ReminderService.ReminderListRefresher");
+            ring = consistentRingProvider;
+            Task.Run(Start).Wait(); // TODO: Lifecycle isn't working, this dirty hack proves reminders work.
```

Comment:
This is temporary because I can't get SiloLifeCycle to fire, which calls `Start()`.

See: https://discord.com/channels/333727978460676096/970765938121130105/970830368620023878